homework/src/index.js

```diff
@@ -31,14 +30,92 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+    const root = document.getElementById('root');
+    const div = createAndAppend('div', root, { class: 'data' });
+    const header = createAndAppend('p', div, { text: 'HYF Repositories', class: 'header' });
+    const table = createAndAppend('table', root, { class: 'table' });
+    const select = createAndAppend('select', header, { id: 'list' });
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody, { text: 'Repository: ', class: 'label' });
+    const repositoryLink = createAndAppend('a', tr);
+    const description = createAndAppend('tr', tbody, { class: 'label' });
+    const forks = createAndAppend('tr', tbody, { class: 'label' });
+    const container = createAndAppend('div', root, { id: 'container' });
+    const updated = createAndAppend('tr', tbody, { class: 'label' });
+
+    fetchJSON(url, (error, repositories) => {
+      if (error) {
+        createAndAppend('div', container, {
+          text: error.message,
+          class: 'alert-error'
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        repositories.forEach((item, index) => {
+          createAndAppend('option', select, { 'text': item.name, 'value': index });
+
+          renderRepositoryBox(
+            repositories[0],
+            repositoryLink,
+            description,
+            forks,
+            updated
+          );
+        });
+      }
+
+      select.addEventListener('change', () => {
+        container.innerHTML = "";
+        const index = event.target.value;
+
+        renderRepositoryBox(
+          repositories[select.value],
+          repositoryLink,
+          description,
+          forks,
+          updated
+        );
+        const contributorOnSelect = repositories[index].contributors_url;
```

Comment:
`contributorOnSelect` is a bit misleading in a name, you could just use `contributors_url`, since that's what the value is: 
```suggestion
        const contributors_url = repositories[index].contributors_url;
```

Reply:
I have changed it.